### PR TITLE
fix: Error handling in UI

### LINKED ISSUE #465: Error handling in UI is broken (ErrorComponent and ErrorBoundary)
**Describe the bug**
Error handling in the UI is broken, a simple use case can be when the API is not reachable and we throw the "KeepApiError", the only thing that is shown on the screen is Keep's logo.

**To Reproduce**
1. Take down the API
2. Start the UI
3. Go to localhost:3000

**Expected behavior**
Error should be presented beautifully with all the context that is possible and a clear CTA should be displayed (e.g. "Try Again")

**Screenshots**
CleanShot 2023-11-08 at 14 30 57

### PR DESCRIPTION
**Pull request by 35C4n0r**
Closes #465 
/claim #465

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: keep-ui/app/error.tsx [hunk 1]
```diff
@@ -1,36 +1,51 @@
...
+  )
 }
 
+// Custom Error Class
 export class KeepApiError extends Error {
   url: string;
+  proposedResolution: string;
 
-  constructor(message: string, url: string) {
+  constructor(message: string, url: string, proposedResolution: string) {
```
Comment (talboren) [inline keep-ui/app/error.tsx:45]: What if it's some other exception, not "KeepApiError"?

**Review by talboren — CHANGES_REQUESTED** [review]
Can you add some documentation on how this works? (e.g., from next website on how Error is handling erros?)

File: keep-ui/app/error.tsx (hunk 1 above)
Comment (35C4n0r) [reply keep-ui/app/error.tsx:52]: Handled that case.

Comment (talboren) [reply keep-ui/app/error.tsx:52]: Can you show how it looks?

Comment (35C4n0r) [reply keep-ui/app/error.tsx:52]: @talboren 
image

**Review by talboren — APPROVED** [review]
LGTM

**Pull request merged**